### PR TITLE
feat: support optional inject

### PR DESCRIPTION
**Pull request by gxkl**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

## Release Notes

- **New Features**
  - Introduced optional dependency injection capabilities, allowing services to be injected as optional.
  - Added a new `InjectOptional` decorator for marking parameters as optional.

- **Improvements**
  - Enhanced error handling in the `EggPrototypeBuilder` for optional inject objects.
  - Streamlined logic for parameter handling in the `Inject` decorator.
  - Improved flexibility in service management by allowing optional dependencies in various services.

- **Tests**
  - Expanded test coverage for optional injections in various scenarios.
  - Added new test cases to validate the functionality of optional dependencies.

- **Documentation**
  - Updated documentation to include detailed sections on optional dependency injection and lifecycle hooks, enhancing clarity and usability.

### CODE REVIEW

**Review by killagu — COMMENTED** [review]
加一下 tegg plugin 和 standalone 的测试。

File: core/core-decorator/src/decorator/Inject.ts [hunk 1]
```diff
@@ -50,3 +56,12 @@ export function Inject(param?: InjectParams | string) {
     }
   };
 }
+
+export function InjectOptional(param?: Omit<InjectParams, 'optional'> | string) {
```
Comment (killagu) [inline core/core-decorator/src/decorator/Inject.ts:60]: Inject optional 为 true 的时候能加类型检查吗？ property 应该为可空。

Comment (gxkl) [reply core/core-decorator/src/decorator/Inject.ts:63]: 限制必须可空的实现方式，报错提示不太直观，暂时先不做类型检查了

**Review by killagu — APPROVED** [review]
LGTM

**Pull request merged**